### PR TITLE
ignore static and declared member if checking override

### LINKED ISSUE #43563: noImplicitOverride complains about static property not overriding non-static property
# Bug Report

### 🔎 Search Terms

- noImplicitOverride
- static

### 🕗 Version & Regression Information

4.3.0-dev.20210406

### 💻 Code

Using `--noImplicitOverride`:

```ts
export class Foo {
    kind = 1
}

class SubFoo extends Foo {
    static kind = '123';
}
```

### 🙁 Actual behavior

On `static kind = 123`, I see:

```
This member must have an 'override' modifier because it overrides a member in the base class 'Foo'.
```

### 🙂 Expected behavior

I don't think this should be an issue. `SubFoo.kind` is distinct from the `Foo.kind` instance property

### LINKED ISSUE #43566: Property using `declare` prompts to add 'override' even though that does not seem valid
# Bug Report

### 🔎 Search Terms

- noImplicitOverride
- static

### 🕗 Version & Regression Information

4.3.0-dev.20210406

### 💻 Code

Using `--noImplicitOverride`:

```ts
export class Foo{
    kind = 1;
}

class SubFoo extends Foo{
    declare kind: undefined;
}
```

### 🙁 Actual behavior

On `declare kind: undefined`, I see:

```
This member must have an 'override' modifier because it overrides a member in the base class 'Foo'
```

Adding `override` makes the code invalid

### 🙂 Expected behavior

We should probably change our code to:

```ts
class SubFoo extends Foo{
    override kind: undefined;
}
```

However using `declare` was previously valid and the current error message seems incorrect

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #43563 
Fixes #43566

Declared property does not really override. So just ignore that.

### CODE REVIEW

**Comment by andrewbranch** [comment]
I think I would expect that under `--noImplicitOverride`, overriding a static property with another static property would require the `override` keyword. The issue with #43563 is that it didn’t seem to recognize that static and instance properties are distinct at all. But maybe I’m missing something?

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Discussed in the design meeting and decided that yes, static overrides need the `override` keyword in `--noImplicitOverride`.

**Review by sandersn — APPROVED** [review]

File: tests/cases/conformance/override/override13.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+class SubFoo extends Foo {
+    property = 42;
+    staticProperty = 42;
+}
+
+class StaticSubFoo extends Foo {
+    static property = 42;
+    static staticProperty = 42;
+}
```
Comment (andrewbranch) [inline tests/cases/conformance/override/override13.ts:17]: Can we add a test that ensures this works more than one heritage clause removed?

```ts
class Base {
    static foo = 0;
}

class Intermediate extends Base {}

class Derived extends Intermediate {
    static foo = 1;
}
```

**Review by andrewbranch — APPROVED** [review]
Thanks @Kingwl!

**Pull request merged**